pytket/extensions/cutensornet/mps/mps.py

```diff
@@ -721,23 +737,37 @@ def get_physical_dimension(self, position: int) -> int:
             self.get_physical_bond(position)
         )
 
-    def copy(self, device_id: Optional[int] = None) -> MPS:
-        """Returns a deep copy of self.
+    def get_device_id(self) -> int:
+        """
+        Returns:
+            The identifier of the device (GPU) where the tensors are stored.
+        """
+        return int(self.tensors[0].data.device)
 
-        Args:
-            device_id: The identifier of the GPU where the copy is meant to be kept.
-                If not provided, the same device as ``self`` will be used.
+    def set_libhandle(self, libhandle: CuTensorNetHandle) -> None:

```

Comment:
Should we allow this to be called in the constructor at all (and pass handle as an optional parameter there)?

Reply:
Yes, done.